src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Mvc.Version1_X/MvcImportProjectFeature.cs

```diff
@@ -68,28 +27,33 @@ @using Microsoft.AspNetCore.Mvc.ViewFeatures
 @addTagHelper Microsoft.AspNetCore.Mvc.Razor.TagHelpers.UrlResolutionTagHelper, Microsoft.AspNetCore.Mvc.Razor
 ");
 
-        private static RazorSourceDocument? s_source;
+    public void CollectImports(RazorProjectItem projectItem, ref PooledArrayBuilder<RazorProjectItem> imports)
+    {
+        ArgHelper.ThrowIfNull(projectItem);
 
-        private DefaultDirectivesProjectItem()
+        // Don't add MVC imports for a component - this shouldn't happen for v1, but just in case.
```

Comment:
Nothing to do with this PR, but man I hate how many 'just in case' comments there are in this codebase

Reply:
💯